### ISSUE TITLE
How to generate resnet50_csv_50_focal_seen_w2v.h5?Hoping to get your reply!

### ISSUE THREAD

**Issue by ZhangTh2** [salman-h-khan/PL-ZSD_Release#12]
Hello,I want to train a model on my own dataset ,but I don`t know how to generate a pre-trained model by using word vectors and focal loss.So can you please support the pre-trained codes,thank you!

**Comment by shafin5** [comment]
I guess you want to generate a pre-trained model with word vectors of seen classes but without using any vocabulary and metric learning. I am attaching the code for it. It can give you an overall idea, but it will not work as it is. Please modify it based on your need. Here I removed the custom layer with an FC layer (last layer before the final layer) and initialized the weight with word vectors. Finally, I made this FC layer non-trainable. You can use such model and train it with losses.focal(). If your unseen classes do not overlap with the unseen classes of MSCOCO, then you can use pre-trained weights of basic RetinaNet to initialize your model partially. I mean only the common layers of your model which is still present in the basic RetinaNet. I hope this answer your question.

[retinanet.txt](https://github.com/salman-h-khan/PL-ZSD_Release/files/3534911/retinanet.txt)

**Comment by gizTM** [comment]
Hello, I have a few questions about this topic. I'm quite new to the field machine learning, and I'm assigned to look into zero shot learning, so sorry if the question is too dumb

How can I train the model with the file you provided with losses.focal()? I downloaded the file you provided above and changed the word vector file path. Changed the import in gzsd.py to the downloaded file. Do I put the code to train the model in gzsd.py and train with losses.focal() instead of loading weight from the pre-trained resnet50_polar_loss.h5? Can I just do that or do I have to train this new polar_loss.h5 separately then run gzsd.py?

I tried to look into keras_retinanet/bin/train_vocab_w2v.py but not sure how to exactly?

**Comment by shafin5** [comment]
I suppose you want to train with focal loss, not polar loss. I that case, you have to modify keras_retinanet/bin/train_vocab_w2v.py file. You should not load any pre-trained file. Also, you have to mention 'classification': losses.focal() at Line 365. For testing you have to run keras_retinanet/bin/evaluate.py. See Running instructions for more detail.

**Comment by gizTM** [comment]
Thank you 👍
